### ISSUE TITLE
Missing accessory type: solar panels and/or solar inverters

### ISSUE THREAD

**Issue by guillochon** [arachnetech/homebridge-mqttthing#560]
Again I have hacked this in as something else for now (an outlet), but it would great to add a proper solar panel type with power produced, etc.

**Comment by rbswift** [comment]
Apple need to support new accessories before they can be added to homebridge plugins

**Comment by x23piracy** [comment]
Hi, you can integrate it as outlet but it will not count negative, is this something the plugin can do for us? Maybe an option to use this outlet negated? (Count it minus).


Best Regards X23

**Comment by arachnetech** [comment]
Mqttthing can only expose HomeKit types supported by Apple. I'm not aware of a Solar Panel/Inverter type.

**Comment by x23piracy** [comment]
yes i know at it works well as outlet but it counting positive, is there a way to interpret the values in negative?
Not from homekit side it just displays what you send over but on mqttthing side? I have no idea if a minus value is possible.

**Comment by arachnetech** [comment]
It might be possible to configure an apply function or codec to invert the value published?

**Comment by x23piracy** [comment]
You mean in mqttthing or the thing sending something to the mqtt broker?
